src/halmos/config.py

```diff
@@ -476,6 +476,13 @@ class Config:
         group=debugging,
     )
 
+    dump_smt_directory: str = arg(
```

Comment:
bring back the `str | None` annotation since the default is None. Or make the default `""`, your choice.